cairo/programs/fork.cairo

```diff
@@ -0,0 +1,203 @@
+    // number
+    assert [range_check_ptr] = header.number;
+    let range_check_ptr = range_check_ptr + 1;
+    // gas_limit
+    assert [range_check_ptr] = header.gas_limit;
+    let range_check_ptr = range_check_ptr + 1;
+    // gas_used
+    assert [range_check_ptr] = header.gas_used;
+    let range_check_ptr = range_check_ptr + 1;
+    // timestamp
+    assert [range_check_ptr] = header.timestamp;
+    let range_check_ptr = range_check_ptr + 1;
+    // mix_hash
+    assert [range_check_ptr] = header.mix_hash.low;
+    let range_check_ptr = range_check_ptr + 1;
+    assert [range_check_ptr] = header.mix_hash.high;
+    let range_check_ptr = range_check_ptr + 1;
+    // nonce
+    assert [range_check_ptr] = header.nonce;
+    let range_check_ptr = range_check_ptr + 1;
+    // base_fee_per_gas
+    assert header.base_fee_per_gas.is_some * (1 - header.base_fee_per_gas.is_some) = 0;
+    assert [range_check_ptr] = header.base_fee_per_gas.value;
+    let range_check_ptr = range_check_ptr + 1;
+    // blob_gas_used
+    assert header.blob_gas_used.is_some * (1 - header.blob_gas_used.is_some) = 0;
+    assert [range_check_ptr] = header.blob_gas_used.value;
+    let range_check_ptr = range_check_ptr + 1;
+    // excess_blob_gas
+    assert header.excess_blob_gas.is_some * (1 - header.excess_blob_gas.is_some) = 0;
+    assert [range_check_ptr] = header.excess_blob_gas.value;
+    let range_check_ptr = range_check_ptr + 1;
+    // parent_beacon_block_root
+    assert header.parent_beacon_block_root.is_some * (
+        1 - header.parent_beacon_block_root.is_some
+    ) = 0;
+    let parent_beacon_block_root = cast(header.parent_beacon_block_root.value, Uint256*);
+    assert [range_check_ptr] = parent_beacon_block_root.low;
+    let range_check_ptr = range_check_ptr + 1;
+    assert [range_check_ptr] = parent_beacon_block_root.high;
+    let range_check_ptr = range_check_ptr + 1;
+    // requests_root
+    assert header.requests_root.is_some * (1 - header.requests_root.is_some) = 0;
+    let requests_root = cast(header.requests_root.value, Uint256*);
+    assert [range_check_ptr] = requests_root.low;
+    let range_check_ptr = range_check_ptr + 1;
+    assert [range_check_ptr] = requests_root.high;
+    let range_check_ptr = range_check_ptr + 1;
+    // extra_data_len
+    assert [range_check_ptr] = header.extra_data_len;
+    let range_check_ptr = range_check_ptr + 1;
+
+    with_attr error_message("InvalidBlock") {
+        assert [range_check_ptr] = header.gas_limit - header.gas_used;
+        let range_check_ptr = range_check_ptr + 1;
+        let expected_base_fee_per_gas = calculate_base_fee_per_gas(
+            header.gas_limit,
+            parent_header.gas_limit,
+            parent_header.gas_used,
+            parent_header.base_fee_per_gas.value,
+        );
+
+        assert expected_base_fee_per_gas = header.base_fee_per_gas.value;
+        assert [range_check_ptr] = header.timestamp - parent_header.timestamp - 1;
```

Comment:
how the parent_header will be filled ?

Reply:
it'll be filled with a hint as well. the currently commented hash computation will ensure that the whole parent header is consistent with the current block